benchmark/gate_kernels.jl

```diff
@@ -0,0 +1,317 @@
+                        (sv = zeros(ComplexF64, $n_amps))
+                suite["gates"][gate_str][(string(n_qubits), string(q), string(q2), "BraketSV")] =
+                    @benchmarkable py_sv.evolve([$py_gate]) setup =
+                        (py_sv = local_sv.StateVectorSimulation($n_qubits, 0, 1))
+            end
+            for q3 in [mod(q+2, n_qubits)] #setdiff(0:n_qubits - 1, q, q2)
+                for (gate_str, gate, py_gate) in zip(
+                    ["CCNot", "CSwap"],
+                    [BraketSimulator.CCNot(), BraketSimulator.CSwap()],
+                    [gate_operations.CCNot([q, q2, 2]), gate_operations.CSwap([q, q2, 2])],
+                )
+                    suite["gates"][gate_str][(
+                        string(n_qubits),
+                        string(q),
+                        string(q2),
+                        string(q3),
+                        "Julia",
+                    )] = @benchmarkable BraketSimulator.apply_gate!($gate, sv, $q, $q2, $q3) setup =
+                        (sv = zeros(ComplexF64, $n_amps))
+                    suite["gates"][gate_str][(
+                        string(n_qubits),
+                        string(q),
+                        string(q2),
+                        string(q3),
+                        "BraketSV",
+                    )] = @benchmarkable py_sv.evolve([$py_gate]) setup =
+                        (py_sv = local_sv.StateVectorSimulation($n_qubits, 0, 1))
+                end
+            end
+        end
+    end
+end
+
+for n_qubits = 2:2:max_qubits(Val(:noise))
+    n_amps = 2^n_qubits
+    prob = 0.1
+    gamma = 0.2
+    for q in 0:0#n_qubits-1
+        for (noise_str, noise, py_noise) in zip(
+            ["BitFlip", "PhaseFlip", "Depolarizing", "AmplitudeDamping", "PhaseDamping"],
+            [
+                BraketSimulator.BitFlip(prob),
+                BraketSimulator.PhaseFlip(prob),
+                BraketSimulator.Depolarizing(prob),
+                BraketSimulator.AmplitudeDamping(prob),
+                BraketSimulator.PhaseDamping(prob),
+            ],
+            [
+                noise_operations.BitFlip([q], prob),
+                noise_operations.PhaseFlip([q], prob),
+                noise_operations.Depolarizing([q], prob),
+                noise_operations.AmplitudeDamping([q], prob),
+                noise_operations.PhaseDamping([q], prob),
+            ],
+        )
+            suite["noise"][noise_str][(string(n_qubits), string(q), "Julia")] =
+                @benchmarkable BraketSimulator.apply_noise!($noise, dm, $q) setup =
+                    (dm = zeros(ComplexF64, $n_amps, $n_amps))
+            suite["noise"][noise_str][(string(n_qubits), string(q), "BraketSV")] =
+                @benchmarkable py_sv.evolve([$py_noise]) setup =
+                    (py_sv = local_dm.DensityMatrixSimulation($n_qubits, 0))
+        end
+        for q2 in [mod(q+1, n_qubits)] #setdiff(0:n_qubits-1, q)
```

Comment:
Is this temporarily commented out?

Reply:
Yes, to avoid doing lots of redundant benchmarks.